File: backend/app/routes/domains.py
"""Routes for domain components."""

from fastapi import APIRouter, HTTPException
from app.data.components import DOMAIN_COMPONENTS

router = APIRouter(prefix="/api/domains", tags=["domains"])


@router.get("/")
async def get_domains():
    """
    Get all available domain components.
    
    Returns all cloud architecture domains with their available components.
    """
    try:
        return DOMAIN_COMPONENTS
    except Exception as e:
        raise HTTPException(status_code=500, detail="Failed to retrieve domains")

File: backend/src/routes/domains.js
// Routes for domain components
const express = require('express');
const router = express.Router();
const DOMAIN_COMPONENTS = require('../data/components');

/**
 * GET /api/domains
 * Retrieve all available domain components
 */
router.get('/', (req, res) => {
  try {
    res.json(DOMAIN_COMPONENTS);
  } catch (error) {
    res.status(500).json({ error: 'Failed to retrieve domains' });
  }
});

module.exports = router;
